src/playback/internal/playbackconfiguration.cpp

```diff
@@ -1,12 +1,129 @@
+#include "Resample.h"
+#include "Dither.h"
+
 #include "playbackconfiguration.h"
 
 using namespace muse;
 using namespace au::playback;
 
+static const std::string moduleName("playback");
+
+static const muse::Settings::Key SOLO_BEHAVIOR(moduleName, "playback/soloBehavior");
+
 draw::Color PlaybackConfiguration::playColor() const
 {
     return "#18A999";
 }
+
+void PlaybackConfiguration::init()
+{
+    muse::settings()->setDefaultValue(SOLO_BEHAVIOR, muse::Val(TracksBehaviors::SoloBehavior::SoloBehaviorMulti));
+    muse::settings()->valueChanged(SOLO_BEHAVIOR).onReceive(nullptr, [this](const muse::Val& val) {
+        m_soloBehaviorChanged.notify();
+    });
+}
+
+std::vector<au::playback::PlaybackQualityPrefs::PlaybackQuality> PlaybackConfiguration::playbackQualityList() const
+{
+    return { PlaybackQualityPrefs::PlaybackQuality::LowQuality, PlaybackQualityPrefs::PlaybackQuality::MediumQuality,
+             PlaybackQualityPrefs::PlaybackQuality::HighQuality, PlaybackQualityPrefs::PlaybackQuality::BestQuality };
+}
+
+au::playback::PlaybackQualityPrefs::PlaybackQuality PlaybackConfiguration::currentPlaybackQuality() const
+{
+    return static_cast<PlaybackQualityPrefs::PlaybackQuality>(Resample::FastMethodSetting.ReadEnum());
+}
+
+void PlaybackConfiguration::setPlaybackQuality(PlaybackQualityPrefs::PlaybackQuality quality)
+{
+    Resample::FastMethodSetting.WriteEnum(static_cast<int>(quality));
+    m_playbackQualityChanged.notify();
+}
+
+std::vector<DitherTypePrefs::DitherType> PlaybackConfiguration::ditheringList() const
+{
+    return { DitherTypePrefs::DitherType::None, DitherTypePrefs::DitherType::Rectangle, DitherTypePrefs::DitherType::Triangle,
+             DitherTypePrefs::DitherType::Shaped };
+}
+
+DitherTypePrefs::DitherType PlaybackConfiguration::currentDithering() const
+{
+    return static_cast<DitherTypePrefs::DitherType>(Dither::FastSetting.ReadEnum());
+}
+
+void PlaybackConfiguration::setDithering(DitherTypePrefs::DitherType dithering)
+{
+    Dither::FastSetting.WriteEnum(static_cast<DitherType>(dithering));
+    m_ditheringChanged.notify();
+}
+
+async::Notification PlaybackConfiguration::ditheringChanged() const
+{
+    return m_ditheringChanged;
+}
+
+async::Notification PlaybackConfiguration::playbackQualityChanged() const
+{
+    return m_playbackQualityChanged;
+}
+
+TracksBehaviors::SoloBehavior PlaybackConfiguration::currentSoloBehavior() const
+{
+    return muse::settings()->value(SOLO_BEHAVIOR).toEnum<playback::TracksBehaviors::SoloBehavior>();
+}
+
+void PlaybackConfiguration::setSoloBehavior(playback::TracksBehaviors::SoloBehavior behavior)
+{
+    muse::settings()->setSharedValue(SOLO_BEHAVIOR, muse::Val(behavior));
+}
+
+muse::async::Notification PlaybackConfiguration::soloBehaviorChanged() const
+{
+    return m_soloBehaviorChanged;
+}
+
+au::trackedit::secs_t PlaybackConfiguration::shortSkip() const
+{
+    double shortSkip;
+    gPrefs->Read(wxT("/AudioIO/SeekShortPeriod"), &shortSkip, 5.0);
```

Comment:
We shouldn't use gPrefs in our configurations, instead of it you should write it to our settings
We registered Au3CommonSettings and if au3 code want to read this pref we will provide the value from our settings

Reply:
same for FastMethodSetting